poetry/utils/env.py

```diff
@@ -318,6 +318,23 @@ def build_venv(cls, path):
 
         build(path)
 
+        # make sure pip and setuptools are on the same version as the base environment
+        from pip import __version__ as pip_version
+        from setuptools import __version__ as setuptools_version
+
+        the_venv = VirtualEnv(Path(path))
+        the_venv.run(
+            "python", "-m", "pip", "install", "--upgrade", "pip=={}".format(pip_version)
+        )
+        the_venv.run(
+            "python",
```

Comment:
This would mean that the creation of new environments would require internet access.

Reply:
Good point. I'll check what can be done. Since we are trying to update to the version available in the base environment there should be some way to get along w/o internet connection.

Reply:
I agree that it seems like there should be a way but the discussion on the python bug tracker seems to think its trickier than it sounds. I hope you find a way though!  I wonder if you should be making a pull request into `ensurepip` instead though.

https://bugs.python.org/issue30628